include/udata.h

```diff
@@ -4,129 +4,187 @@
+
     /** absolute tolerances for integration */
-    double atol;
+    double atol = 1e-16;
+
     /** relative tolerances for integration */
-    double rtol;
+    double rtol = 1e-8;
+
     /** maximum number of allowed integration steps */
-    int maxsteps;
+    int maxsteps = 0;
 
     /** maximum number of allowed Newton steps for steady state computation */
-    int newton_maxsteps;
+    int newton_maxsteps = 0;
+
     /** maximum number of allowed linear steps per Newton step for steady state
      * computation */
-    int newton_maxlinsteps;
-    /** Preequilibration of model via NEwton solver? */
-    int newton_preeq;
+    int newton_maxlinsteps = 0;
+
+    /** Preequilibration of model via Newton solver? */
+    int newton_preeq = false;
+
     /** Which preconditioner is to be used in the case of iterative linear
      * Newton solvers */
-    int newton_precon;
+    int newton_precon = 1;
 
-    /** internal sensitivity method */
-    /*!
-     * a flag used to select the sensitivity solution method. Its value can be
-     * CV SIMULTANEOUS or CV STAGGERED. Only applies for Forward Sensitivities.
-     */
-    int ism;
+    /** internal sensitivity method flag used to select the sensitivity solution
+     * method. Its value can be CV SIMULTANEOUS or CV STAGGERED. Only applies
+     * for Forward Sensitivities. */
+    int ism = 1;
 
     /** method for sensitivity computation */
-    AMICI_sensi_meth sensi_meth;
+    AMICI_sensi_meth sensi_meth = AMICI_SENSI_FSA;
 
     /** linear solver specification */
-    int linsol;
+    int linsol = 9;
 
-    /** interpolation type */
-    /*!
-     * specifies the interpolation type for the forward problem solution which
+    /** interpolation type for the forward problem solution which
      * is then used for the backwards problem. can be either CV_POLYNOMIAL or
      * CV_HERMITE
      */
-    int interpType;
+    int interpType = 1;
 
-    /** linear multistep method */
-    /*!
-     * specifies the linear multistep method and may be one of two possible
+    /** specifies the linear multistep method and may be one of two possible
      * values: CV ADAMS or CV BDF.
      */
-    int lmm;
+    int lmm = 2;
 
-    /** nonlinear solver */
-    /*!
+    /**
      * specifies the type of nonlinear solver iteration and may be either CV
      * NEWTON or CV FUNCTIONAL.
      */
-    int iter;
+    int iter = 2;
 
     /** flag controlling stability limit detection */
-    booleantype stldet;
+    booleantype stldet = true;
 
-    /** state initialisation */
-    double *x0data;
+    /** state initialisation (size np) */
+    double *x0data = nullptr;
 
-    /** sensitivity initialisation */
-    double *sx0data;
+    /** sensitivity initialisation (size nx * nplist) */
+    double *sx0data = nullptr;
 
     /** state ordering */
-    int ordering;
-    
-    /** function to print the contents of the UserData object */
-    void print();
+    int ordering = 0;
```

Comment:
options are AMD: 0 COLAMD: 1 natural:2 I don't think we have corresponding defines yet